src/python/gudhi/point_cloud/knn.py

```diff
@@ -90,6 +96,10 @@ def fit(self, X, y=None):
         Args:
             X (numpy.array): coordinates for reference points.
         """
+        if self.k > len(X):
+            raise ValueError(
```

Comment:
Raising an error certainly makes sense. I am trying to think if there could be other useful behaviors. For instance, if we compute the DTM-filtration of many point sets with the same `k` and one of the point sets happens to have fewer than `k` points, what should we do? Probably lower `k` to the number of points (not sure what else we could do if we don't want an error). But then where/how should we lower `k`, in knn? in dtm? in weightedrips? in dtmfiltration? in the caller of dtmfiltration?
Just leaving this comment here so I can think about it at a better time, but probably raising an error here as you did is good.